lib/ansible/modules/cloud/vmware/vmware_datastore_maintenancemode.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_datastore_maintenancemode
+short_description: Place a datastore into maintenance mode
+description:
+    - This module can be used for to manage maintenance mode of datastore.
+author:
+- "Abhijeet Kasurde (@akasurde)"
+version_added: 2.6
+notes:
+    - Tested on vSphere 5.5, 6.0 and 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    datastore:
+      description:
+      - Name of datastore to manage.
+      - If C(datastore_cluster) or C(cluster_name) are not set, this parameter is required.
+    datastore_cluster:
+      description:
+      - Name of datastore cluster from all child datastores to be managed.
+      - If C(datastore) or C(cluster_name) are not set, this parameter is required.
+    cluster_name:
+      description:
+      - Name of cluster from all child datastores to be managed.
```

Comment:
`Name of the cluster from which all child datastores need to be managed`

I find this a bit weird, btw... clusters do not really have datastores directly as children, only datacenters have datastores directly: You can not create a datastore specific to a cluster, you create datastores on the datacenter, and connect hosts to them. I assume the `cluster.datastores` just lists all datastores that are connected to hosts in that cluster.

Personally, it would make more sense to use datacenter as a grouping parameter instead of cluster (but that might also be a bit excessive, as that would put all datastores in maintenance mode in a DC, basically preventing any VM to run at all...)

Reply:
I kept this option as user may want to manage datastores which are related to a particular cluster. This can be one of the use case. But, I am OK for removing this option altogether, if it is introducing confusion.

Reply:
Just leave it, it is in line with other modules...

Reply:
Changed wording of description.